lib/ews/soap.rb

```diff
@@ -27,11 +27,21 @@ module SOAP
       NS_SOAP         = 'soap'.freeze
       NS_EWS_TYPES    = 't'.freeze
       NS_EWS_MESSAGES = 'm'.freeze
+      NS_AUTODISCOVER = 'a'.freeze
+      NS_ADDRESING    = 'wsa'.freeze
+      NS_SCHEMA       = 'xsi'.freeze
       NAMESPACES = {
         "xmlns:#{NS_SOAP}"         => 'http://schemas.xmlsoap.org/soap/envelope/',
         "xmlns:#{NS_EWS_TYPES}"    => 'http://schemas.microsoft.com/exchange/services/2006/types',
         "xmlns:#{NS_EWS_MESSAGES}" => 'http://schemas.microsoft.com/exchange/services/2006/messages',
       }.freeze
+      AUTODISCOVER_NAMESPACES = {
+        "xmlns:#{NS_AUTODISCOVER}"  => 'http://schemas.microsoft.com/exchange/2010/Autodiscover',
+        "xmlns:#{NS_ADDRESING}"     => 'http://www.w3.org/2005/08/addressing',
+        "xmlns:#{NS_SCHEMA}"        => 'http://www.w3.org/2001/XMLSchema-instance',
+        "xmlns:#{NS_SOAP}"          => 'http://schemas.xmlsoap.org/soap/envelope/'
+      }.freeze
+      GET_USER_SETTINGS_ACTION = "http://schemas.microsoft.com/exchange/2010/Autodiscover/Autodiscover/GetUserSettings".freeze
```

Comment:
maybe `GET_USER_SETTINGS_ACTION_URL` is better name as when reviewing the `build_get_user_settings_soap ` it reads a bit confusing - probably just my own problem 😂